packages/core/src/indexing/database-index-api.ts

```diff
@@ -155,34 +156,24 @@ export class DatabaseIndexApi<DateType = Date | number> {
   /**
    * This method inserts the stream if it is not present in the index, or updates
    * the 'content' if the stream already exists in the index.
-   * @param args
+   * @param indexingArgs
    */
-  async indexStream(args: IndexStreamArgs & { createdAt?: Date; updatedAt?: Date }): Promise<void> {
-    const tableName = asTableName(args.model)
-    await this.indexDocumentInDatabase(tableName, args)
-  }
-
-  private async indexDocumentInDatabase(
-    tableName: string,
+  async indexStream(
     indexingArgs: IndexStreamArgs & { createdAt?: Date; updatedAt?: Date }
   ): Promise<void> {
+    const tableName = asTableName(indexingArgs.model)
     const indexedData = this.getIndexedData(indexingArgs)
-    for (const field of this.modelsIndexedFields.get(indexingArgs.model.toString()) ?? []) {
+    const fields = this.modelsIndexedFields.get(indexingArgs.model.toString()) ?? []
+    for (const field of fields) {
       indexedData[addColumnPrefix(field)] = indexingArgs.streamContent[field]
     }
-
     await this.dbConnection(tableName).insert(indexedData).onConflict('stream_id').merge({
       last_anchored_at: indexedData.last_anchored_at,
       updated_at: indexedData.updated_at,
+      stream_content: indexedData.stream_content,
```

Comment:
Don't we need to also add the extra fields to index, in case they are updated?
Maybe the logic could simply be `this.dbConnection(tableName).insert(indexedData).onConflict('stream_id').merge(indexedData)` to ensure no data is missing?

Reply:
Ohh, true.